src/librustdoc/clean/mod.rs

```diff
@@ -275,7 +273,7 @@ impl Clean<WherePredicate> for hir::WherePredicate<'_> {
             hir::WherePredicate::BoundPredicate(ref wbp) => {
                 let bound_params = wbp
                     .bound_generic_params
-                    .into_iter()
+                    .iter()
```

Comment:
Why this change?

Reply:
Lint. But it didn't make much sense I guess to consume the type here I guess?

Reply:
I'm asking *your* opinion, not clippy's opinion. I don't think it's useful to change code just because clippy says to, there should be a reason.

In particular, this prevents reusing the underlying allocation, which uses more memory than it did before.

Reply:
So it's a regression. Reverting it then.

The full story is actually that clippy was linting about the `collect` (which can't be removed unfortunately). After a few tries, I simply ended with this and didn't revert completely, my bad.